addons/l10n_ch/models/res_bank.py

```diff
@@ -117,12 +114,12 @@ def _l10n_ch_get_qr_vals(self, amount, currency, debtor_partner, free_communicat
             '0200',                                               # Version
             '1',                                                  # Coding Type
             acc_number,                                           # IBAN / QR-IBAN
-            'K',                                                  # Creditor Address Type
+            'S',                                                  # Creditor Address Type
             (self.acc_holder_name or self.partner_id.name)[:70],  # Creditor Name
-            creditor_addr_1,                                      # Creditor Address Line 1
-            creditor_addr_2,                                      # Creditor Address Line 2
-            '',                                                   # Creditor Postal Code (empty, since we're using combined addres elements)
-            '',                                                   # Creditor Town (empty, since we're using combined addres elements)
+            self.partner_id.street,                               # Creditor Address Line 1
+            self.partner_id.street2 or '',                        # Creditor Address Line 2
```

Comment:
Maybe it should be more like:
```suggestion
            self.partner_id.street,                               # Creditor Street
            '',                                                   # Creditor Building number
```

Other option would be to use `_get_street_split`:
```python
    street_data = self.partner_id._get_street_split()
    
    # (...)
            (street_data.get('street_name') or self.partner_id.street)[:70],  # Creditor Street
            (street_data.get('street_number) or '')[:16],                     # Creditor Building number